language/vm/vm-runtime/src/code_cache/module_cache.rs

```diff
@@ -34,9 +35,24 @@ use vm_runtime_types::{
 #[cfg(test)]
 use crate::code_cache::module_adapter::FakeFetcher;
 
-/// Trait that describe a cache for modules. The idea is that this trait will in charge of
-/// loading resolving all dependencies of needed module from the storage.
-pub trait ModuleCache<'alloc> {
```

Comment:
Shall we just keep the trait just so that we can have a place to specify the loader api? I don't mind if we just change the trait definitions to include `InterpreterContext` or `ChainState` as needed.

Reply:
as I said I really don't want to keep any trait around if traits are not needed.
I understand we ay want to reintroduce some later but I'd rather wait for that abstraction to be understood rather than to push a trait just because.
I am not sure what the loader API will look like and I want maximum simplicity which I also believe is something we should all strive for.
I found extremely difficult to work with our code base and I feel we have a lot of abstraction that have no reason to exist.
Again I would love to see the story between `ChainState` and `InterpreterContext` being formalized better. `RemoteCache` should go. And I am not how many traits we really need